File: 算法/1. 两数之和/Solution.go
package ___两数之和

func twoSum(nums []int, target int) []int {
	memo := make(map[int]int)
	for key, value := range nums {
		result := target - value
		if value, ok := memo[result]; ok {
			return []int{value, key}
		}
		memo[value] = key
	}
	return []int{}
}

File: 算法/1. 两数之和/Solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> memo = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int result = target - nums[i];
            if (memo.containsKey(result)) {
                return new int[]{memo.get(result), i};
            }
            memo.put(nums[i], i);
        }
        return new int[0];
    }

}
